### PR TITLE
Rename schema to database in PostgresHook

### PR DESCRIPTION
**Pull request by feluelle**
In PostgresHook the "schema" field is only being called like that to make it compatible with the underlying DbApiHook which uses the schema for the sql alchemy connector. The postgres connector library however does not allow setting a schema in the connection instead a database can be set. To clarify that, we change all references in the PostgresHook code and documentation.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow-providers-postgres/connections/postgres.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Host (required)
     The host to connect to.
 
 Schema (optional)
-    Specify the schema name to be used in the database.
+    Specify the database name.
```
Comment (feluelle) [inline docs/apache-airflow-providers-postgres/connections/postgres.rst]: Note that the field in the connection will still be `schema`. This is due to how we build the connection string for all databases. But I updated the docs to make sure that it should contain the database name.

File: airflow/providers/postgres/hooks/postgres.py [hunk 2]
```diff
@@ -95,7 +100,7 @@ def get_conn(self) -> connection:
             host=conn.host,
             user=conn.login,
             password=conn.password,
-            dbname=self.schema or conn.schema,
+            dbname=self.database or conn.schema,
```
Comment (kaxil) [inline airflow/providers/postgres/hooks/postgres.py:107]: Do we still want to set `dbanem` from `conn.schema`?

Comment (kaxil) [reply airflow/providers/postgres/hooks/postgres.py:107]: Is this mainly for backwards-compat?

File: docs/apache-airflow-providers-postgres/connections/postgres.rst (hunk 1 above)
Comment (kaxil) [reply docs/apache-airflow-providers-postgres/connections/postgres.rst]: Should we add a reason in the docs as well on why we are using `schema` field to set `database`?

File: airflow/providers/postgres/hooks/postgres.py (hunk 2 above)
Comment (kaxil) [reply airflow/providers/postgres/hooks/postgres.py:107]: The following should work for setting schema?
```
postgresql://localhost/airflow?options=-csearch_path%3Ddbo,public
```

Comment (feluelle) [reply airflow/providers/postgres/hooks/postgres.py:107]: Interesting 🤔 I will test that.

File: docs/apache-airflow-providers-postgres/connections/postgres.rst (hunk 1 above)
Comment (feluelle) [reply docs/apache-airflow-providers-postgres/connections/postgres.rst]: Thanks @kaxil I have updated the docs. PTAL :)

File: airflow/providers/postgres/hooks/postgres.py (hunk 2 above)
Comment (feluelle) [reply airflow/providers/postgres/hooks/postgres.py:107]: I don't want to add a new feature for adding this in the connection. We can add that separately if really necesseary. 

I have documented how users can set the default schema as an additional query or use PostgresOperator runtime params to run the searchpath option before the queries get executed.

**Review by feluelle — COMMENTED** [review]
I added backwards compatibility. Please review again @kaxil. 🙏

File: airflow/providers/postgres/operators/postgres.py [hunk 3]
```diff
@@ -42,6 +42,8 @@ class PostgresOperator(BaseOperator):
         (default value: False)
     :param parameters: (optional) the parameters to render the SQL query with.
     :param database: name of database which overwrite defined one in connection
+    :param runtime_parameters: a mapping of runtime params added to the final sql being executed.
+        E.g you could set the schema via {"search_path": "CUSTOM_SCHEMA"}
```
Comment (feluelle) [inline airflow/providers/postgres/operators/postgres.py]:
```suggestion
        For example, you could set the schema via `{"search_path": "CUSTOM_SCHEMA"}`.
```

File: airflow/providers/postgres/hooks/postgres.py [hunk 4]
```diff
@@ -67,10 +68,17 @@ class PostgresHook(DbApiHook):
     supports_autocommit = True
 
     def __init__(self, *args, **kwargs) -> None:
+        if 'schema' in kwargs:
+            warnings.warn(
+                'The "schema" arg has been renamed to "database" as it contained the database name.'
+                'Please use "database" to set the database name.',
+                DeprecationWarning,
```
Comment (feluelle) [inline airflow/providers/postgres/hooks/postgres.py:75]:
```suggestion
                DeprecationWarning,
                stacklevel=2,
```

File: airflow/providers/postgres/hooks/postgres.py (hunk 2 above)
Comment (potiuk) [reply airflow/providers/postgres/hooks/postgres.py:107]: Agree. I think it's a simple/clever way of breaking of the schema/database naming issues :)

File: docs/apache-airflow-providers-postgres/connections/postgres.rst [hunk 5]
```diff
@@ -29,7 +29,7 @@ Host (required)
     The host to connect to.
 
 Schema (optional)
-    Specify the schema name to be used in the database.
+    Specify the name of the database to connect to. Note that a database schema is not allowed during connect. Instead you can use ``SET search_path TO custom_schema`` if you want to change the default schema being used in a session for your queries. If you are using ``PostgresOperator`` see :ref:`Passing Server Configuration Parameters into PostgresOperator <howto/operators:postgres>`.
```
Comment (potiuk) [inline docs/apache-airflow-providers-postgres/connections/postgres.rst]: One comment here. I think it needs a bit more explanation here. What does it mean for the user that "database schema is not allowed during connect"  - I think it should be described a bit more from the user perspective: "If you do not use Postgres operator but run query directly via Postgres Hook, you need also to set  schema via "SET search_path". ... Or comething like that (if I understand correctly the consequences)

Comment (feluelle) [reply docs/apache-airflow-providers-postgres/connections/postgres.rst]: I have updated the docs in 8e9b1ba to reference postgres docs. I think this makes it clearer & cleaner. WDYT?

Comment (potiuk) [reply docs/apache-airflow-providers-postgres/connections/postgres.rst]: Looks better :)

**Review by potiuk — APPROVED** [review]
LGTM. WDYT @kaxil ?

File: airflow/providers/postgres/hooks/postgres.py [hunk 6]
```diff
@@ -67,10 +68,18 @@ class PostgresHook(DbApiHook):
...
+                stacklevel=2,
+            )
+            kwargs['database'] = kwargs['schema']
```
Comment (ashb) [inline airflow/providers/postgres/hooks/postgres.py:78]: Please can you add a test that ensures this still works

Comment (feluelle) [reply airflow/providers/postgres/hooks/postgres.py:78]: Done. PTAL :)

**Review by kaxil — APPROVED** [review]

**Pull request merged**